sessions/01-exploration-tools/README.md

```diff
@@ -0,0 +1,714 @@
+# 0x01. Exploration Tools
+## Resources
+
+TODO@slides
+
+TODO@"session archive" link
+
+## Tutorials
+When faced with a binary with no source or parts of the source missing you can infer some of its functionalities based upon some basic reconnaissance techniques using various tools.
+### 01. Tutorial - Poor man's technique: strings
+The simplest recon technique is to dump the ASCII (or Unicode) text from a binary. It doesn't offer any guarantees but sometimes you can get a lot of useful information out of it.
+
+>By default, when applied to a binary it only scans the data section. To obtain information such as the compiler version used in producing the binary use `strings -a`.
+
+Let's illustrate how strings can be useful in a simple context. Try out crackme1 from the tutorial archive (01-tutorial-strings directory):
+
+```c
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+ 
+int my_strcmp(char *s1, char *s2)
+{
+	size_t i, len = strlen(s1);
+    if (len == 0)
+	return -1;
+    for (i = 0; i < len; i++)
+        if (s1[i] != s2[i])
+	    return -1;
+    return 0;
+}
+
+int main()
+{
+    char buf[1000];
+ 
+    printf("Password:\n");
+    if (fgets(buf, 1000, stdin) == NULL)
+	exit(-1);
+
+    buf[strlen(buf) - 1] = '\0';
+
+    if (!my_strcmp(buf, ???????????????????????)) {
+        printf("Correct!\n");
+    } else
+	printf("Nope!\n");
+
+    return 0;
+}
+```
+
```

Comment:
Should we add the command used to compile the code? I know most of them should know that by now...